### PR TITLE
Delay setting content root until after build is called.

### PR DESCRIPTION
**Pull request by jkotalik**
Besides adding a new main that will call WebHostBuilder.UseContentRoot(), is there any way to write a test for this?

### CODE REVIEW

**Comment by pakrym** [comment]
You can extract this logic into a class and mock `iisConfigData`.

File: src/Microsoft.AspNetCore.Server.IISIntegration/WebHostBuilderIISExtensions.cs [hunk 1]
```diff
@@ -56,9 +56,10 @@ public static IWebHostBuilder UseIISIntegration(this IWebHostBuilder hostBuilder
                     throw exception;
                 }
 
-                hostBuilder.UseContentRoot(iisConfigData.pwzFullApplicationPath);
                 return hostBuilder.ConfigureServices(services =>
                 {
+                    // Delay setting the content root so users don't accidentally overwrite it.
+                    hostBuilder.UseContentRoot(iisConfigData.pwzFullApplicationPath);
```
[the remark this answers is not in the document]
Comment (pakrym) [reply src/Microsoft.AspNetCore.Server.IISIntegration/WebHostBuilderIISExtensions.cs]: Why doesn't `ConfigureHostConfiguration` work, it runs just before creating HostingEnvironment.

[the remark this answers is not in the document]
Comment (Tratcher) [reply src/Microsoft.AspNetCore.Server.IISIntegration/WebHostBuilderIISExtensions.cs]: Oooh, ConfigureHostConfiguration sounds promising.

**Comment by jkotalik** [comment]
Closing for now. This approach won't work and I think we will need to design a new way of doing this.

**Pull request closed without merging**